File: frontend_web/src/services/tokenService.ts
function getPayload(): Record<string, unknown> | null {
  const token = localStorage.getItem('jwt_token')
  if (!token) return null
  try {
    const payload = JSON.parse(atob(token.split('.')[1]))
    if (payload.exp * 1000 < Date.now()) return null
    return payload
  } catch {
    return null
  }
}

export function getUserId(): string | null {
  const p = getPayload()
  if (!p) return null
  const id =
    p['sub'] ??
    p['nameid'] ??
    p['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier']
  return id ? String(id) : null
}

export function getUserName(): string | null {
  const p = getPayload()
  if (!p) return null
  const name =
    p['name'] ??
    p['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name']
  return name ? String(name) : null
}

File: frontend_mobile/lib/services/token_service.dart
import 'package:flutter_secure_storage/flutter_secure_storage.dart';
import 'package:jwt_decoder/jwt_decoder.dart';

class TokenService {
  static const _storage = FlutterSecureStorage();

  static Future<String?> getUserId() async {
    final token = await _storage.read(key: 'jwt_token');
    if (token == null) return null;
    if (JwtDecoder.isExpired(token)) return null;
    final payload = JwtDecoder.decode(token);
    final id = payload['sub'] ??
        payload['nameid'] ??
        payload['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier'];
    return id?.toString();
  }

  static Future<String?> getUserName() async {
    final token = await _storage.read(key: 'jwt_token');
    if (token == null) return null;
    if (JwtDecoder.isExpired(token)) return null;
    final payload = JwtDecoder.decode(token);
    return (payload['name'] ??
            payload['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name'])
        ?.toString();
  }
}
